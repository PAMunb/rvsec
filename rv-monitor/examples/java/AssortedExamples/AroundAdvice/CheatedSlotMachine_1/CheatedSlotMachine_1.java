package CheatedSlotMachine_1;

import rvm.CheatedSlotMachineRuntimeMonitor;

import casino.SlotMachine;

public class CheatedSlotMachine_1 {
	public static void main(String[] args){
		SlotMachine machine = new SlotMachine();
		for (int i = 0; i < 10 ; ++i) {
			System.out.println("Round " + i);
			machine.insertCoin();
			rvm.CheatedSlotMachineRuntimeMonitor.insert_coinEvent(machine);
			machine.push();
			rvm.CheatedSlotMachineRuntimeMonitor.push_buttonEvent(machine);
			rvm.CheatedSlotMachineRuntimeMonitor.resultEvent(machine);
			if (!CheatedSlotMachineRuntimeMonitor.skipEvent)
				System.out.println(machine.getResult());
		}
	}
}
