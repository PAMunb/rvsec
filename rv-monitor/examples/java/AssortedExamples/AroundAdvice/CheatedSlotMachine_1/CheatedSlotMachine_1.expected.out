Round 0
73
Round 1
41
Round 2
Round 3
Round 4
Round 5
Round 6
Round 7
Round 8
Round 9
